crates/api_models/src/payments.rs

```diff
@@ -1013,6 +1013,17 @@ pub struct PaymentsRequest {
     /// Choose what kind of sca exemption is required for this payment
     #[schema(value_type = Option<ScaExemptionType>)]
     pub psd2_sca_exemption_type: Option<api_enums::ScaExemptionType>,
+
+    /// Service details for external authentication
+    #[schema(value_type = Option<ServiceDetails>)]
+    pub service_details: Option<ServiceDetails>,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct ServiceDetails {
+    pub merchant_transaction_id: Option<String>,
+    pub correlation_id: Option<String>,
+    pub x_src_flow_id: Option<String>,
```

Comment:
Can you add documentation for all the fields ? 
NIT: Name `x_src_flow_id` better